### PR TITLE
Test modeling rules junit

### PR DESCRIPTION
**Pull request by kobymeir**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-6879
Content: https://github.com/demisto/content/pull/29085

## Description
Test modeling rules junit

### CODE REVIEW

**Comment by daryakoval** [comment]
Hi, Please do not merge without my approval. I am a code owner of these files, and I want to review them as well.

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (DeanArbel) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:73]: Since there are a few constants being added to the global scope, the next step would be to extract them to a relative constants file.

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (DeanArbel) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:78]: Why did you remove the return type hint?

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (DeanArbel) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:224]: Please consider extracting the code in the loop to another function.

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (DeanArbel) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:542]: nice

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (DeanArbel) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:644]: Consider extracting this to a helper function.

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (kobymeir) [reply demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:78]: fixed

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (kobymeir) [reply demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:224]: done

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (kobymeir) [reply demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:644]: done

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (tkatzir) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:982]: Consolidate to a single log

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (tkatzir) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:983]: Why not log instead of `echo`?

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (tkatzir) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:1013]: docstring?

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (tkatzir) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:1048]: Why not `logger.info` instead of `echo`?

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (tkatzir) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:1210]:
```suggestion
        f"[cyan]Modeling Rules directories to test: {inputs}[/cyan]",
```

**Review by tkatzir — APPROVED** [review]
Very good!
Approved, with minor comments.

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (kobymeir) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:83]:
```suggestion
        String representation of a table, to display the expected and received values.
```

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (kobymeir) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:302]:
```suggestion
    if expected_values:  # fixme, invert if
```

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (kobymeir) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:310]:
```suggestion
        typer.echo(table_result)  # fixme logger.info
```

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (kobymeir) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:546]:
```suggestion
) -> TestCase:  # fixme docstring, ret value
```

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (kobymeir) [reply demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:605]: i don't understand what's missing

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (kobymeir) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:668]:
```suggestion
        system_out = "Test data pushed successfully"  # fixme add test data id
```

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (kobymeir) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:682]:
```suggestion
) -> bool:  # fixme rename to is_pack_exists_on_server
```

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (kobymeir) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:759]:
```suggestion
            first list: ids that do not have example event data.  # fixme rename function is_test_data_exists_on_server
```

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (kobymeir) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:783]:
```suggestion
) -> Tuple[bool, Union[TestSuite, None]]:  # fixme docstring return value
```

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (kobymeir) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:842]:
```suggestion
        logger.info(  # fixme merge log lines
```

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (kobymeir) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:1012]:
```suggestion
                if modeling_rule.testdata_path:  # fixme merge log lines
```

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (kobymeir) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:1054]:
```suggestion
def handle_missing_event_data_in_modeling_rule(  # fixme docstring
```

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py
Comment (tkatzir) [reply demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:610]: In line 606 (formerly 485) you note that the Dataset doesn't exist after X seconds.
In line 596 (formerly 476) you on't mention the number of seconds.

So either add them in line 596, or do it here, to cover both cases.

Comment (kobymeir) [reply demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:614]: fixed

**Pull request merged**
